test/integration/controller/jobset_controller_test.go

```diff
@@ -64,11 +64,6 @@ var _ = ginkgo.Describe("JobSet validation", func() {
 		}, timeout, interval).Should(gomega.BeTrue())
 	})
 
-	ginkgo.AfterEach(func() {
-		// Delete test namespace after each test.
-		gomega.Expect(k8sClient.Delete(ctx, ns)).To(gomega.Succeed())
-	})
```

Comment:
as I mentioned on the issue, I think it is better to have a DeleteNamespace func that lists the deletes the objects we know get created as part of those integration tests, we did that in kueue: https://github.com/kubernetes-sigs/kueue/blob/main/test/util/util.go#L79

Reply:
Sounds good.  I'll add that.